slashing-protection/src/main/java/tech/pegasys/web3signer/slashingprotection/dao/SignedAttestationsDao.java

```diff
@@ -28,56 +29,36 @@
       final int validatorId,
       final UInt64 targetEpoch,
       final Bytes signingRoot) {
-
-    final Query query;
-    if (signingRoot == null) {
-      query =
-          handle
-              .createQuery(
-                  "SELECT validator_id, source_epoch, target_epoch, signing_root "
-                      + "FROM signed_attestations WHERE (validator_id = ? AND target_epoch = ?) AND (signing_root IS NOT NULL)")
-              .bind(0, validatorId)
-              .bind(1, targetEpoch);
-    } else {
-      query =
-          handle
-              .createQuery(
-                  "SELECT validator_id, source_epoch, target_epoch, signing_root "
-                      + "FROM signed_attestations WHERE (validator_id = ? AND target_epoch = ?) AND (signing_root <> ? OR signing_root IS NULL)")
-              .bind(0, validatorId)
-              .bind(1, targetEpoch)
-              .bind(2, signingRoot);
-    }
-
-    return query.mapToBean(SignedAttestation.class).list();
+    checkNotNull(signingRoot, "This function only accepts queries where the signing root is known");
```

Comment:
Seems a bit odd to say "This function only accept queries" that's a very developer specific message. Perhaps just something like "Signing root must non-null"?

Reply:
If we fail a "checkNotNull", it means we have a bug (i.e. this is low level parameter verification) - so its going to need a developer - having said that, happy to fix.